containers/CategoriesContainer.js

```diff
@@ -0,0 +1,101 @@
+import { Input, Form, FormGroup } from 'reactstrap';
+import { useState, useEffect } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+import ResourcesService from '../services/resources-service';
+import Category from '../components/Category';
+import ValidationError from '../errors/validation-error';
+
+const CategoriesContainer = ({ categories, setCategories, setShowAddWordIcon, wordlistEntryId }) => {
+  if (!categories) throw new ValidationError('categories should be an Array');
+
+  const [showAddCategoriesInput, setShowAddCategoriesInput] = useState(false);
+  const [newCategoryNames, setNewCategoryNames] = useState();
+
+  useEffect(() => {
+    if (!showAddCategoriesInput) return;
+
+    const input = document.getElementById('categories-input');
+
+    input.addEventListener('blur', () => {
+      setShowAddCategoriesInput(false);
+      setShowAddWordIcon(true);
+    });
+
+    input.addEventListener('keydown', ({ key }) => {
+      if (key !== 'Escape') return;
+
+      setShowAddCategoriesInput(false);
+      setShowAddWordIcon(true);
+    });
+  }, [showAddCategoriesInput]);
+
+  const handleChange = ({ target: { value: name }}) => setNewCategoryNames(name.trim());
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const newNames = newCategoryNames.split(',').map(i => i.trim().toLowerCase()).filter(i => i.length);
+    if (categories.map(i => i.name).some(name => newNames.includes(name))) return; // need to display some sort of warning message instead of just returning
```

Comment:
just put a little icon (little red circle, white cross in middle) with "duplicate category" next to it, underneath the input box

Reply:
or better: just don't add it